daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilInputContentHandler.scala

```diff
@@ -0,0 +1,250 @@
+      // we started another element while we were in the process of building a startElement
+      // this means the first element was complex and we are ready for the inputter queue
+      sendToInputter()
+    }
+    // use Attributes to determine xsi:nil value
+    val nilIn = atts.getIndex(XMLConstants.W3C_XML_SCHEMA_INSTANCE_NS_URI, "nil")
+    infosetEvent.nilValue = if (nilIn >= 0) {
+      val nilValue = atts.getValue(nilIn)
+      One(nilValue)
+    } else {
+      Nope
+    }
+    // set localName and namespaceURI
+    val qNameArr = qName.split(":")
+    infosetEvent.localName =
+      if (localName.nonEmpty) {
+        One(localName)
+      } else if (qNameArr.length > 1) {
+        qNameArr.lift(1)
+      } else if (qNameArr.nonEmpty) {
+        qNameArr.lift(0)
+      } else {
+        Nope
+      }
+    infosetEvent.namespaceURI =
+      if (uri.nonEmpty) {
+        One(uri)
+      } else if (qNameArr.length > 1) { // has a prefix
+        // get the prefix off the qname
+        val qNamePrefix = qNameArr.lift(0).get
+        // look up prefix with and without xmlns prefix
+        val in = atts.getIndex(s"xmlns:$qNamePrefix")
+        val otherIn = if (in == -1) atts.getIndex("qNamePrefix") else in
+        if (in >= 0) {
+          val attrUri = atts.getValue(in)
+          One(attrUri)
+        } else if (otherIn >= 0) {
+          val attrUri = atts.getValue(otherIn)
+          One(attrUri)
+        } else {
+          One(prefixMapping.getURI(qNamePrefix))
+        }
+      } else {
+        try {
+          val in = atts.getIndex("xmlns")
+          if (in >= 0) {
+            val attrUri = atts.getValue(in)
+            One(attrUri)
+          } else {
+            One(prefixMapping.getURI(null))
+          }
+        } catch {
+          case _: NullPointerException => Nope
+        }
+      }
+    infosetEvent.eventType = One(StartElement)
+  }
+
+  override def endElement(uri: String, localName: String, qName: String): Unit = {
+    // if infosetEvent is a startElement, send that first
+    if (infosetEvent.eventType.contains(StartElement)) {
+      // any characterData that exists at this point is valid data as padding data has been
+      // taken care of in startElement
+      val maybeNewStr = One(characterData.toString())
+      infosetEvent.simpleText = maybeNewStr
+      characterData.setLength(0)
+      sendToInputter()
+    }
+    val qNameArr = qName.split(":")
+    infosetEvent.localName =
+      if (localName.nonEmpty) {
+        One(localName)
+      } else if (qNameArr.nonEmpty) {
+        One(qNameArr.last)
+      } else {
+        Nope
+      }
+    infosetEvent.namespaceURI =
+      if (uri.nonEmpty) {
+        One(uri)
+      } else if (qNameArr.length > 1) {
+        One(prefixMapping.getURI(qNameArr.head))
+      } else {
+        try {
+          One(prefixMapping.getURI(null))
+        } catch {
+          case _: NullPointerException => Nope
```

Comment:
If prefixMapping.getURI(null) can throw a NPE, then can the call to prefixMapping.getURI above also? So should this try/catch surround that also?